plugins/playstation/src/playstation-plugin.vala

```diff
@@ -0,0 +1,40 @@
+// This file is part of GNOME Games. License: GPLv3
+
+private class Games.PlayStation : Object, Plugin {
+	private const string FINGERPRINT_PREFIX = "playstation";
+	private const string MIME_TYPE = "application/x-cue";
+	private const string MODULE_BASENAME = "libretro-playstation.so";
+	private const bool SUPPORTS_SNAPSHOTTING = true;
+
+	public GameSource get_game_source () throws Error {
+		var query = new MimeTypeTrackerQuery (MIME_TYPE, game_for_uri);
+		var connection = Tracker.Sparql.Connection.@get ();
+		var source = new TrackerGameSource (connection);
+		source.add_query (query);
+
+		return source;
+	}
+
+	private static Game game_for_uri (string uri) throws Error {
+		var cue_file = File.new_for_uri (uri);
+		var cue_sheet = new CueSheet (cue_file);
+		var cue_track_node = cue_sheet.get_track (0);
+		var cue_file_node = cue_track_node.file;
+		var bin_file = cue_file_node.file;
```

Comment:
It's OK to chain up non nullable fields and properties, so you can do `cue_track_node.file.file` if you want, but it's not that big of a deal, what you did is fine too. :)